### PR TITLE
MM-41332 : Move "SERVER_DIR" env var to cli args for emoji_gen file

### PR DESCRIPTION
**Pull request by scottaudet**
#### Summary

Updated emoji_gen file to accept cli param for path to mattermost-server instead of checking for environment var

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/19434

#### Related Pull Requests
N/A

#### Release Note

```release-note
npm command "make-emojis" now accepts server-dir argument for path to server.
```

### CODE REVIEW

File: build/emoji/emoji_gen.mjs
Comment (M-ZubairAhmed) [inline build/emoji/emoji_gen.mjs:35]:
```suggestion
    .example('npm run $0 -- --server-dir ../mattermost-server', 'path to mattermost-server for copying emoji_data.go file')
```

File: build/emoji/emoji_gen.mjs
Comment (M-ZubairAhmed) [inline build/emoji/emoji_gen.mjs:37]: its path of mattermost server
```suggestion
        description: 'Path to mattermost-server',
```

File: build/emoji/emoji_gen.mjs
Comment (M-ZubairAhmed) [inline build/emoji/emoji_gen.mjs:381]: `eslint-disable-next-line no-process-env` can be removed since we arent using the env

File: build/emoji/emoji_gen.mjs
Comment (M-ZubairAhmed) [inline build/emoji/emoji_gen.mjs:380]: lets extract `argv['server-env']` to a constant as in `argsExcludedEmojiFile`

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]
Looks good. Also extremely sorry i wrote at first `server-env` before the argument to be now i realized `server-dir` is a better argument name. Would you be open in changing that as well? thank you :)

File: Makefile
Comment (scottaudet) [reply Makefile:155]: That makes sense! Would it be nice to make this npm run make-emojis $(ARGUMENTS) so they can be passed to this make command if they desire?

File: build/emoji/emoji_gen.mjs
Comment (scottaudet) [reply build/emoji/emoji_gen.mjs:13]: Thanks for catching this. This definitely should not have been changed

**Comment by M-ZubairAhmed** [comment]
please re request review once you are ready :)

File: build/emoji/emoji_gen.mjs
Comment (M-ZubairAhmed) [inline build/emoji/emoji_gen.mjs:21]: nice job with this `assert {type: 'json'}` but is this out of box supported in node version we are using? i see it is still stage 3

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

**Comment by scottaudet** [comment]
All changes have been made - thanks for the feedback @M-ZubairAhmed!

File: Makefile
Comment (M-ZubairAhmed) [inline Makefile:155]: I have tried running this locally. Do you know how we can pass the arguments via make. Both of the below doesnt seem to work.
1. `make emojis --server-dir ../mattermost-server`
image

2. `make emojis -- --server-dir ../mattermost-server`
image

Actually should this command be in the make file anyways @hmhealey ?

Comment (scottaudet) [reply Makefile:155]: make ARGUMENTS='--server-dir ../mattermost-server' emojis

Comment (hmhealey) [reply Makefile:155]: > Actually should this command be in the make file anyways @hmhealey ?

Yeah, let's keep it here for now. It might be nice to get rid of the Makefile one day to be more JS-friendly, but there's currently a lot of stuff here that doesn't fit in the package.json, so that'd require somewhere else to put that.

**Review by M-ZubairAhmed — APPROVED** [review]
🆗  It looks good to me. Thank you

**Comment by cwarnermm** [comment]
We're moving the contents of https://github.com/mattermost/docs/pull/5400 to the User Forum. Once this PR is merged, please ensure Docs is notified so that we can update that User Forum post with these enhancements (which simplify the user's effort). Thanks!

**Comment by M-ZubairAhmed** [comment]
@cwarnermm yes will definitely inform the Docs once this gets merged. Will also update the google doc with the updated information

**Review by Willyfrog — APPROVED** [review]
LGTM! added a small comment, but I don't think it should block

**Comment by M-ZubairAhmed** [comment]
Adding security review since it contains changes to `Makefile`

**Review by oh6hay — APPROVED** [review]
LGTM

**Comment by hmhealey** [comment]
Updating this again to see if that fixes the upload-s3 CI step. I'm not sure why that would be failing at the moment, but it's likely not related to any changes made in this PR.

**Comment by M-ZubairAhmed** [comment]
i am going to try to remove the changes to makefile and see if it ci upload passes.

**Pull request merged**
